feat: team selection

MLopezJ opened the pull request:
Add a modal on the game view to allow user to select the team they want to be part of. This way, the user is going to see in color others robots and be able to set custom gesture of her/his team, and in black color and without any action available any robot else.

Current implementation means admin needs to set the teams name and attach each robot with its team.

And improvement for this feature is to use local storage instead of hook to save the team of the user.

MLopezJ committed:
feat: create Modal component

Create a modal component for the app.

MLopezJ committed:
style: update positioning of class

MLopezJ committed:
refactor: move state related to manage the list of the teams name options

Previously that state was on TeamAssigner component, which makes dificult to access it on others components.

New implementation move it into useGameController hoock

MLopezJ committed:
feat: select team

Make the users able to select the team they are gonna be part of

MLopezJ committed:
feat: select team to play with

On game view there is a modal to request the user to select the team

MLopezJ committed:
fix: show modal to select team when team has not been selected

MLopezJ committed:
fix: remove nested ternary operator

MLopezJ committed:
feat: add default oponent robot color as a constant

MLopezJ committed:
fix: set correct types on CameControllerContext definition

expose a new method from GameControllerContext to update local state. This way, the  type definition can be updated as well and 'any' was removed.

As part of the changes, varaibles were renamed.

MLopezJ committed:
fix: add default team name options

KajaKoren commented on src/components/Admin/TeamAssigner.tsx:
```diff
@@ -62,7 +57,8 @@ export const TeamAssigner: FunctionComponent = () => {
 				className="btn btn-danger"
 				onClick={() => {
-					setStateOpt([{ name: 'A' }, { name: 'B' }])
+					addTeamNameOption('A')
```
Should not add, but reset to team "A" and "B"

Lenakh97 committed:
fix: outline the robots not selected

Lenakh97 committed:
fix: use admin shadow to list 'select team' options (#28)

Other changes includes changing the name of a component from 'Modal' to 'Popup' and from 'SelectTeam' to 'TeamSelectionUser'.

Co-authored-by: Kaja Koren <kaja.koren@nordicsemi.no>
Co-authored-by: Mauro Lopez <mauro.lopez@nordicsemi.no>
